### PR TITLE
OCRVS-73: Add birth event json form

### PR DESCRIPTION
**Pull request by napiergit**

### CODE REVIEW

File: app/scripts/directives/add-events/add-event/forms/birth.json [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        "value": null
+      }, {
+        "type": "input",
+        "flex": "20",
+        "name": "babyMiddleNames",
+        "title": "Baby Middle Names",
+        "settings": {
+          "valueType": "valueString",
+          "disabled": false,
+          "required": true
```
Comment (BMartinos) [inline app/scripts/directives/add-events/add-event/forms/birth.json]: i think we can set this to false, not all babies will have a middle name

File: app/scripts/directives/add-events/add-event/forms/birth.json [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        "value": null
+      }, {
+        "type": "input",
+        "flex": "20",
+        "name": "mothersMiddleNames",
+        "title": "Mothers Middle Names",
+        "settings": {
+          "valueType": "valueString",
+          "disabled": false,
+          "required": true
```
Comment (BMartinos) [inline app/scripts/directives/add-events/add-event/forms/birth.json]: Im not sure if we really need this field? but if we do, the required property should be false as not all people have middle names

File: app/scripts/controllers/add-events.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ module.exports = function (Api, $scope, $routeParams, $location) {
   Api.Patients.get({ id: $routeParams.patientId }, success, error)
 
   $scope.events = [
-    { code: 'sample-event', display: 'Sample Event', formName: 'FormBuilderaddEventSampleEvent' }
+    { code: 'birth', display: 'Birth', formName: 'FormBuilderAddEventBirth' }
```
Comment (BMartinos) [inline app/scripts/controllers/add-events.js]: Lets update the display name to "Birth Registration" to give more of a description what this event is form, Same applies to the form heading

File: app/scripts/directives/add-events/add-event/forms/birth.json [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        },
+        "skipLogic": {
+          "func": {},
+          "checks": []
+        },
+        "value": null
+      }, {
```
Comment (BMartinos) [inline app/scripts/directives/add-events/add-event/forms/birth.json]: Lets put the date and place of birth on a new formbuilder row

File: app/scripts/directives/add-events/add-event/forms/birth.json [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+          "disabled": false,
+          "required": true
+        },
+        "skipLogic": {
+          "func": {},
+          "checks": []
+        },
+        "value": null
+      },{
+        "type": "input",
```
Comment (BMartinos) [inline app/scripts/directives/add-events/add-event/forms/birth.json]: Lets put the mothers details on a new formbuilder row

File: app/scripts/controllers/add-events.js [hunk 6]
```diff
@@ -35,7 +35,7 @@ module.exports = function (Api, $scope, $routeParams, $location) {
   Api.Patients.get({ id: $routeParams.patientId }, success, error)
 
   $scope.events = [
-    { code: 'sample-event', display: 'Sample Event', formName: 'FormBuilderaddEventSampleEvent' }
+    { code: 'birth', display: 'Birth Registration', formName: 'FormBuilderAddEventBirth' }
```
Comment (rcrichton) [inline app/scripts/controllers/add-events.js]: Just a note, this should be 'Birth Notification'. The terminology is interesting here. In CRVS:

* a notification is basically a flag to say a birth happened with some basic info and someone needs to follow up to get a full declaration
* a declaration is the submission of all birth details with mother and father identity document and any other required info
* a registration is when a birth declaration is verified by some official and is considered registered, at this point a birth certificate can be issued

Comment (BMartinos) [inline app/scripts/controllers/add-events.js]: Thanks @rcrichton thats a very good explanation of the differences!, So for now, We are ONLY dealing with Birth Notification?

**Comment by BMartinos** [comment]
Thanks @napiergit looking good! Just the one comment mentioned by Ryan, And one last one from my side is to update the version of FormBuilder so that we can include the inputTelephoneNumber control that includes the number validator
https://github.com/jembi/sssa-frontend/blob/master/app/resources/forms/register/emergencyContactInfo.json#L39-L46

**Comment by BMartinos** [comment]
Thanks @napiergit LGTM :+1:

File: app/scripts/controllers/add-events.js (hunk 6 above)
Comment (rcrichton) [inline app/scripts/controllers/add-events.js]: @BMartinos that's right.

**Comment by kaweesi** [comment]
hi @BMartinos did you notice the failing tests; here is my log; https://gist.github.com/kaweesi/2e6ab23369fb175071718f16f2477e5f

**Comment by BMartinos** [comment]
Hi @kaweesi Yip these tests would be failing as this ticket doesnt include the document being sent to hearth, that will be handled in OCRVS-74.

**Pull request merged**